core/src/banking_stage.rs

```diff
@@ -2108,4 +2203,181 @@ mod tests {
+            assert_eq!(packets_vec.len(), num_conflicting_transactions);
+            for single_packets in &packets_vec {
+                assert_eq!(single_packets.packets.len(), 1);
+            }
+            let mut buffered_packets: UnprocessedPackets = packets_vec
+                .clone()
+                .into_iter()
+                .map(|single_packets| (single_packets, vec![0]))
+                .collect();
+
+            let (continue_sender, continue_receiver) = unbounded();
+            let (finished_packet_sender, finished_packet_receiver) = unbounded();
+
+            let test_fn = Some(move || {
+                finished_packet_sender.send(()).unwrap();
+                continue_receiver.recv().unwrap();
+            });
+            // When the poh recorder has a bank, should process all non conflicting buffered packets.
+            // Processes one packet per iteration of the loop
+            let interrupted_iteration = 1;
+            poh_recorder.lock().unwrap().set_bank(&bank);
+            let poh_recorder_ = poh_recorder.clone();
+            let (gossip_vote_sender, _gossip_vote_receiver) = unbounded();
+            // Start up thread to process the banks
+            let t_consume = Builder::new()
+                .name("consume-buffered-packets".to_string())
+                .spawn(move || {
+                    BankingStage::consume_buffered_packets(
+                        &Pubkey::default(),
+                        &poh_recorder_,
+                        &mut buffered_packets,
+                        None,
+                        &gossip_vote_sender,
+                        test_fn,
+                        None,
+                    );
+
+                    // Check everything is correct. All indexes after `interrupted_iteration`
+                    // should still be unprocessed
+                    assert_eq!(
+                        buffered_packets.len(),
+                        packets_vec[interrupted_iteration + 1..].iter().count()
+                    );
+                    for ((remaining_unprocessed_packet, _), original_packet) in buffered_packets
+                        .iter()
+                        .zip(&packets_vec[interrupted_iteration + 1..])
+                    {
+                        assert_eq!(
+                            remaining_unprocessed_packet.packets[0],
+                            original_packet.packets[0]
+                        );
+                    }
+                })
+                .unwrap();
+
+            for i in 0..=interrupted_iteration {
+                finished_packet_receiver.recv().unwrap();
+                if i == interrupted_iteration {
+                    poh_recorder
+                        .lock()
+                        .unwrap()
+                        .schedule_dummy_max_height_reached_failure();
+                }
+                continue_sender.send(1).unwrap();
```

Comment:
this `1` also can be replaced with `()`? hehe, my laziness here. ;) My other comment maybe didn't mention this.

Reply:
... and please remember this: https://github.com/solana-labs/solana/pull/15210#discussion_r573865727 :)

Reply:
Changed, same comment as: https://github.com/solana-labs/solana/pull/15210#discussion_r574875850